server/lib/PLS/Server/Request/TextDocument/PublishDiagnostics.pm

```diff
@@ -151,9 +151,16 @@ sub get_compilation_errors
     push @setup, (chdir => $new_cwd) if (length $new_cwd and -d $new_cwd);
 
     my @diagnostics;
+    my @loadfile;
+    if ($suffix && $suffix eq 'pm') {
```

Comment:
```suggestion
    if (length $suffix and $suffix eq 'pm') {
```

Reply:
Does this need to look at .pmc files too? What about .al?

Reply:
I'm not familiar with those suffixes, so you tell me, i guess?

Reply:
Perl looks at .pmc files before .pm files when requiring. I'm not sure what a .pmc file even is though.

.al files are AutoLoader files, something I'm also not familiar with.

I think, just for ease, we should change the if statement to

```
if (not length $suffix or $suffix eq 'pl')
{
   @loadfile = (-c => $path);
}
else
{
    @loadfile = (-e => "BEGIN { require '$path' }");
}
```

Reply:
One last concern - if someone pathological puts a single quote in their file path, this will break. Should probably do

```
$path =~ s/'/\\'/g
```

Before using.

Reply:
> Perl looks at .pmc files before .pm files when requiring. I'm not sure what a .pmc file even is though.
> 
> .al files are AutoLoader files, something I'm also not familiar with.
> 
> I think, just for ease, we should change the if statement to
> 
> ```
> if (not length $suffix or $suffix eq 'pl')
> {
>     @loadfile = (-c => $path);
> }
> else
> {
>     @loadfile = (-e => "BEGIN { require '$path' }");
> }
> ```

It looks to me like pmc files are a totally dead format, and I don't even know if we WOULD handle them correctly either way.

I just experimented with .al files. They look like normal packages, so I suppose we should support them

Reply:
> One last concern - if someone pathological puts a single quote in their file path, this will break. Should probably do
> 
> ```
> $path =~ s/'/\\'/g
> ```
> 
> Before using.

that's not a problem: PLS always checks against a temporary file, not the original (as far as I understood)

Reply:
The file is created in the same directory as the original file.

Reply:
i implemented this safety thingy by using `\Q` and `\E` in the interpolated -e, which will quote any non letter/number for the interpolated string.

Reply:
actually, the \Q thing didn't work.
I did, however, test a file that had a ' in its name. I didn't have any problems even without doing the replacement. Are you sure we need to?

Reply:
ah, nevermind, it would only happen if the dir has the '. my bad